### DISCUSSION TITLE
Most/least recent repos

### DISCUSSION THREAD

**Discussion by RussKie** [gitextensions/gitextensions#9858] [Q&A]
Follow up on https://github.com/gitextensions/gitextensions/issues/4942#issuecomment-1022637372 and below:

![image](https://user-images.githubusercontent.com/6248932/151252825-ac91ce5c-6595-4630-ad98-3dcc023e257e.png)

@RussKie:

> My view is the concept of "most/least recent" should be retired and removed. The Recent and Favourites lists provide sufficient functionality to organise the collection of repositories...

@gerhardol:

> > My view is the concept of "most/least recent" should be retired and removed. The Recent and Favourites lists provide sufficient functionality to organise the collection of repositories.
> 
> most/least recent gives me about what I need, I do not want to manually handle the 50+ repos I work with (that is with worktrees, without submodules so in reality there is >= 100 repos).

. . .

I'd like to learn how "most/least recent repositories" are used and what tangible benefit they have over the "recent" and the "favourites" lists.

**Comment by gerhardol** [comment]
I work with maybe 20 different separate main repos. A little simplified, each of them create a separate binary with a specific functionality.
For many of them I also use worktrees, so there are maybe 60 "main repos" in my main clone folder.
Most projects are in windows, some recent in wsl (and some in vmware, not with GE).
These main repos have three to ten submodules each.  The submodules have the actual applications (that are developed separately from the "binary containers") but also build scripts, common functionality in all and some functionality shared just in a few main repos. So most submodules are used in many "binaries".
This is embedded, so the build chain is very complex so the binary can be optimized. There is a high cost to be able to use cheap hardware.

I mostly work with the shared parts and therefore have to work with many of the main repos to for instance make changes for submodule changes.
A normal week I open at least 50 separate main or subrepos. Some weeks there are more but more than 50 repos in the list is too long.
If using one main repo, I use the side panel to switch repos, but most recent is convenient to jump from one main repo submodule to another.

Using favorites is simply too much maintenance, main repos are added and removed dynamically. The "grouping" of projects is also dynamic. Either all repos are in one group or the group constantly changes. Submodules could be in separate groups to main repos.
To only add superprojects would simplify, but I do use the most recent for submodules and this would be a quite hidden setting.

**Comment by mstv** [comment]
I use only a different limited number of repos on the different PCs. `Sort alphabetically` works best for me.
I prefer to open separate GE instances using the Jump List of the taskbar icon or using a script.
There is no need to use the Dashboard for me.
The `Favourite Repositories` with its subsubsubmenu is unusable IMO.

When I might need to work with Google Repo (which is ignoring submodules) more heavily, it might become unwieldy - both the lists of repos and - more feared - viewing all the diffs belonging to a commit in the meta repo.
